[FEATURE] Add events for dev server handling

sirdiego opened the pull request:
Hi folks, thanks for the great work.

We are interested in working on this. This is just a PR so we can work on this together.

Since Vite version 7.1.0 Vite resolves the server URIs before `listening` event gets called (https://github.com/vitejs/vite/pull/19981). With this we can create a "lock" file when the dev server starts, that contains the dev servers URI. We use this [Vite plugin](https://www.npmjs.com/package/@hdnet/vite-plugin-typo3-lock) to create `var/vite.lock` and can read the server URI at runtime from there.

This way we:
1. Do not have to hardcode any server URI
2. Can work locally without the dev server running if needed

If you are interested in adding events or something like that, please reach out.

Thanks!

With this PR we could hook into the **vite-asset-colletor** like this:

```php
#[AsEventListener(identifier: 'vite-check-server', event: UseDevServerEvent::class, method: 'check')]
#[AsEventListener(identifier: 'vite-get-server', event: GetDevServerEvent::class, method: 'get')]
final class ViteDevServerListener
{
    public function check(UseDevServerEvent $event): void
    {
        try {
            $content = \file_get_contents(Environment::getVarPath() . DIRECTORY_SEPARATOR . 'vite.lock');
            if ($content === false) {
                return;
            }

            $uri = \json_decode($content);
            new Uri($uri);
            $event->useDevServer = true;
        } catch (\Throwable) {
            $event->useDevServer = false;
        }
    }

    public function get(GetDevServerEvent $event): void
    {
        try {
            $content = \file_get_contents(Environment::getVarPath() . DIRECTORY_SEPARATOR . 'vite.lock');
            if ($content === false) {
                return;
            }

            $uri = \json_decode($content);
            $event->uri = (string)(new Uri($uri));
        } catch (\Throwable) {
        }
    }
}
``` 

Our `vite.config.js` looks like this:
```javascript
export default defineConfig({
// ...
plugins: [typo3(), lock('var/vite.lock')],
// ...
})
```

sirdiego committed:
[FEATURE] Add events for dev server handling

sirdiego committed:
[TASK] Change event names and add tests

s2b commented:
Sorry about my naming details, but can we use `ModifyUseDevServerEvent`? That way the naming would be consistent.

Other than that, this looks great. And thanks for adding those test cases. I'd expect to potentially move these events around with an upcoming refactoring, so those will be very helpful!

sirdiego committed:
[TASK] Refactor event structure

This allows the user of the event to use the original configuration
value and the internally resolved value based on the configuration.

s2b commented:
Hm, now it gets more complicated and I'm not sure if I like that. Apart from the confusing naming, I'm not sure if these events can remain stable after the refactoring happened. For me, these look like two events now: One before and one after the resolution of `auto`.

Can't we keep it simple for now and start with a small event that covers the most common use case?
